src/model.js

```diff
@@ -0,0 +1,242 @@
+const yelp = require("yelp-fusion");
+const proj4 = require("proj4");
+const _ = require("lodash");
+const async = require("async");
+
+const {
+  koopProviderYelp: { api_key },
+} = require("config");
+const client = yelp.client(api_key);
+const YELP_API_MAX_RADIUS = 40000;
+
+const proj = proj4("GOOGLE", "WGS84");
+
+function Model() {}
+
+// each model should have a getData() function to fetch the geo data
+// and format it into a geojson
+Model.prototype.getData = function (req, callback) {
+  if (req.query && req.query.returnCountOnly) {
+    return callback(null, {
+      type: "FeatureCollection",
+      count: 2001,
+      features: [],
+    });
+  }
+  const queries = buildQueries(req.query);
+  const featureCollection = {
+    type: "FeatureCollection",
+    features: [],
+  };
+
+  async.each(queries, search, finish);
+
+  function search(query, cb) {
+    // If this service is added to an ArcGIS Online Map, usually it makes multiple
+    // calls (4 quadrants). When you make 4 calls to the Yelp API at the same
+    // time, you get a Yelp error, so for now set a random pause 0.5-2 seconds.
+    // TODO: Remove this random pause code, and change it so that if we get a
+    // "TOO_MANY_REQUESTS_PER_SECOND" error, retry after a random pause.
```

Comment:
I think you should probably do this now instead of random timer.  Maybe use exponential backoff.

Reply:
I disagree with the `exponential backoff`. All requests are valid and we still want them to finish as soon as possible (within limit).

Leaking bucket is a better solution.

Reply:
@haoliangyu good thinking, totally right